### PR TITLE
chore: separate lint workflow

### PR DESCRIPTION
**Pull request by ignatiusmb**
We may not know if CI actually failed or flaked out at a glance, separating lint into its own job would make it easier to read.

### CODE REVIEW

**Comment by ignatiusmb** [comment]
Whoops, I changed the matrix version array to just numbers because we only care about the major versions and it's recommended to use this format instead. Are the 'required' checks manually assigned to "12.x" and "14.x" strings or something, is there some way to adjust those?

**Comment by ignatiusmb** [comment]
While we're here, we should replace them (the checks) with v16 — arguably v18 too — as v12 is already EOL and v14 is nearing to EOL in 7 months.

**Comment by robertsLando** [comment]
> Are the 'required' checks manually assigned to "12.x" and "14.x" strings or something, is there some way to adjust those?

Unfortunally yes and I don't have the privileges to change this

**Comment by robertsLando** [comment]
> While we're here, we should replace them (the checks) with v16 — arguably v18 too — as v12 is already EOL and v14 is nearing to EOL in 7 months.

We also support some EOL versions

**Comment by ignatiusmb** [comment]
> We also support some EOL versions

Could you please explain why?

**Comment by ignatiusmb** [comment]
> and I don't have the privileges to change this

Don't mean to be rude, but do we know anyone we can or should mention for this?

**Comment by leerob** [comment]
CleanShot 2022-09-24 at 10 50 54@2x
Updated

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,11 +7,21 @@ on:
...
+          node-version: 16
+      - run: yarn install
+      - run: yarn lint
+
   test:
     strategy:
       fail-fast: false # prevent test to stop if one fails
       matrix:
-        node-version: [12.x, 14.x, 16.x, 18.x]
+        node-version: [12, 14, 16, 18]
```
Comment (jesec) [inline .github/workflows/ci.yml:24]: No need for this change. Otherwise LG2M.

**Review by jesec — CHANGES_REQUESTED** [review]

**Review by robertsLando — APPROVED** [review]

**Comment by ignatiusmb** [comment]
What's left to do?

**Review by robertsLando — APPROVED** [review]

**Comment by robertsLando** [comment]
@ignatiusmb Could you sync this with main branch please?

**Review by robertsLando — APPROVED** [review]

**Comment by robertsLando** [comment]
Unfortunately I cannot merge this as there is Jesec review peding and I cannot dismiss it anyway... The only one who can is @leerob or we could simply close and reopen the pr

**Comment by baparham** [comment]
> Unfortunately I cannot merge this as there is Jesec review peding and I cannot dismiss it anyway... The only one who can is @leerob or we could simply close and reopen the pr

I was able to dismiss it

**Pull request closed without merging**